Add DIP-39 (#138)
efdevcon/DIPs: DIPs/DIP-38.md

---

> DIP: 38

Reviewer: ```suggestion
DIP: 39
```

we need to use 39 as 38 is already taken by the fileverse DIP

> Title: Integration of zkPassport for Thai Citizen Discount Verification

Reviewer: ```suggestion
Title: Integration of zkPassport for SEA builder discounts
```

not only Thai here - but all of SEA

Status: Draft
Themes: Ticketing, Purchases & ID
Instances: [Devcon7]
Tags: Software
Authors: Michael Elliot (@michaelelliot), Théo Madzou (@madztheo)
Created: 2024-06-08
---

## Summary of Proposal

> This proposal introduces an zkPassport integration for Devcon in Bangkok, allowing Thai citizens to prove their nationality using zero-knowledge proofs, ensuring privacy and security while accessing ticket discounts.

Reviewer: ```suggestion
This proposal introduces an zkPassport integration for Devcon in Bangkok, allowing SEA citizens to prove their nationality using zero-knowledge proofs, ensuring privacy and security while accessing ticket discounts.
```

## Abstract

> The integration enables Thai citizens to use zkPassport to generate a zero-knowledge proof of their nationality without revealing additional personal information, facilitating a secure and private way to obtain discounted tickets.

Reviewer: ```suggestion
The integration enables citizens of SEA to use zkPassport to generate a zero-knowledge proof of their nationality without revealing additional personal information, facilitating a secure and private way to obtain discounted tickets.
```

## Motivation & Rationale

> To provide a privacy-preserving method for Thai citizens to claim Devcon ticket discounts, avoiding the need for sharing sensitive identity documents. This method ensures only Thai citizens can access the discount, using zero-knowledge proofs to protect personal information.

Reviewer: ```suggestion
To provide a privacy-preserving method for citizens of SEA to claim Devcon ticket discounts, avoiding the need for sharing sensitive identity documents. This method ensures only citizens of SEA can access the discount, using zero-knowledge proofs to protect personal information.
```

## Implementation
**High Level User Flow**:
1. Users scan their ePassport via NFC into the zkPassport app.
2. On the Devcon ticket ordering site, users click a button to generate a unique QR code containing the credentials to prove (their nationality) and a callback URL.
3. Users scan this QR code, which opens the zkPassport app and prompts the user to prove the requested credentials (their nationality).
4. The user accepts, and a proof is generated locally on their phone, and submitted to the callback URL.
5. Backend verifies the proof, checks the `discount_codes` table, and returns a discount code if valid.

**Circuit Public Inputs**:

> - Nationality (expected to be THA)

Reviewer: ```suggestion
- Nationality (expected to be in the set of SEA countries)
```
